src/dc_array.rs

```diff
@@ -42,13 +42,7 @@ pub unsafe fn dc_array_free_ptr(array: *mut dc_array_t) {
     }
     let mut i: size_t = 0i32 as size_t;
     while i < (*array).count {
-        if (*array).type_0 == 1i32 {
-            free(
-                (*(*(*array).array.offset(i as isize) as *mut _dc_location)).marker
-                    as *mut libc::c_void,
-            );
-        }
-        free(*(*array).array.offset(i as isize) as *mut libc::c_void);
+        Box::from_raw(*(*array).array.offset(i as isize) as *mut dc_location);
```

Comment:
This is probably right since @dignifiedquire didn't comment on this in the original PR.  But for my understanding.  This is supposed to free the memory by the deallocator of the type that's being boxed.  In this case that's `*mut uintptr_t` IIUC.  Where's Drop defined for the raw pointer to uintptr_t?

Reply:
`Box::from_raw(*mut dc_location)` here constructs a `Box<dc_location>`